kratos/geometries/triangle_3d_3.h

```diff
@@ -903,8 +903,17 @@ template<class TPointType> class Triangle3D3
         CoordinatesArrayType point_projected = rPoint;
         const array_1d<double,3> vector_points = rPoint - r_center.Coordinates();
         const double distance = inner_prod(vector_points, normal);
-        if (std::abs(distance) > std::numeric_limits<double>::epsilon()) { // Not in the plane, projecting
-            noalias(point_projected) = rPoint - normal * distance;
+
+        // We check if we are on the plane
+        if (std::abs(distance) > std::numeric_limits<double>::epsilon()) {
+            if (std::abs(distance) > 5.0e-1 * Length()) {

```

Comment:
Perhaps we should be more consistent with the base class implementation, i.e., use a similar tolerance and issue a warning.

Reply:
The proper thing would be refactor current tests. The tests are supposed to run with a small tolerance of this

Reply:
Ok, if the tests are the problem maybe we should fix those instead. If there is another reason for putting this through now we should clearly document it with a comment since it's not obvious.

Reply:
Yes, I think in this PR more like a PR/discussion

Reply:
Can you point out which tests are problematic?

Reply:
Mostly contact and mortar related(but some others too). Before me not many people used the PointLocalCoordinates in the Triangle3D3N, that's why the original implementation was mine

Reply:
I don't like this o.5_*length 
Seems to be a sum of other problem which reflects here

Reply:
I know, this PR is to materialize a future PR that will actually solve the issue, probably changing many tests. So with this now people will be aware of the issue.

Reply:
My opinion would be to use a tolerance similar to the base class Newton Raphson. Returning some large distance (100 in this case) as a way to signal failure to the caller is not consistent with the base since the base doesn't make any guarantees.

Reply:
I know, but the previous method was using an iterative approximation, then hidding potential errors (the only thing was checked was the convergence of the iterative algorithm). This PR is not to merge, is to show that the current implementation is wrong, so in a future real solution PR people will understand the origin of the change.

Reply:
What course of action do you suggest?

Reply:
Consider a more realistic value and check one by one the tests until everything works fine

Reply:
how abt the following steps to not have another eternal PR:
1. You put a value you think is suitable
2. The others can then test their things in this branch
3. After a certain period of time (idk, 4 weeks?) this will be merged

what do you think @KratosMultiphysics/technical-committee @loumalouomega ?

Reply:
More or lees what I propose, but most of the tests are mine

Reply:
More I look into it less I like the idea... IMO if you want to check if you are in plane or in certain distance of plane, you should do this by distance method rather than changing the local coordinate system. This should be in client part leaving the local coordinate system as smooth as it was with all of its expected good or bad behaviour.